### ISSUE TITLE
Remove mock APIs and replace with real implementations

### ISSUE THREAD

**Issue by jnordberg** [steemit/yo#54]

**Comment by GarethNelson** [comment]
The mock API already is being replaced, the test flag is there for testing
external apps and removing it will only slow down development.

**Comment by GarethNelson** [comment]
I've labelled this wontfix for now, but i'm open to arguments for removing the mock APIs - my view is that doing so will slow down development work and give us nothing in return.

Without the mock API in place, every test of external apps must work with real data, and every time we want to verify behaviour of an API call we must do so manually as there's nothing to compare it to.

**Comment by jnordberg** [comment]
What I'm getting at is having the API actually implemented, if you think it would be faster to keep the mock implementation in parallel that's fine

**Comment by GarethNelson** [comment]
With the original title this wasn't clear as to the meaning, it's also more of a meta issue tracking the various API features.

Anyway, i've self-assigned and will update/close once it appears the API is feature-complete with the real DB backing it.
